examples/src/main/java/io/grpc/examples/header/HeaderClientInterceptor.java

```diff
@@ -0,0 +1,37 @@
+package io.grpc.examples.header;
+
+import io.grpc.*;
+
+import java.util.logging.Logger;
+
+/**
+ * @author zhaohaifeng
+ * @since 2015-03-16
+ */
+public class HeaderClientInterceptor implements ClientInterceptor {
```

Comment:
A point to mention: MetadataUtils can be used if the cookie isn't going to change over time. If the cookie may be refreshed then creating your own interceptor like this can work better.